src/pool.rs

```diff
@@ -377,7 +386,11 @@ impl ConnectionPool {
     /// Ban an address (i.e. replica). It no longer will serve
     /// traffic for any new transactions. Existing transactions on that replica
     /// will finish successfully or error out to the clients.
-    pub fn ban(&self, address: &Address, shard: usize) {
+    pub fn ban(&self, address: &Address, shard: usize, process_id: i32) {
```

Comment:
It's hard to explain why `ban` would need `process_id` until you read the function code. Maybe it's better to move the stats handling to the caller?